web/crux/src/app/deploy/deploy.http.controller.ts

```diff
@@ -1,52 +1,174 @@
 import JwtAuthGuard, { IdentityFromRequest } from '../token/jwt-auth.guard'
+import {
+  CreateDeploymentDto,
+  DeploymentDetailsDto,
+  DeploymentDto,
+  DeploymentEventDto,
+  InstanceDto,
+  InstanceSecretsDto,
+  PatchDeploymentDto,
+  PatchInstanceDto,
+} from './deploy.dto'
 import DeployService from './deploy.service'
-import DeployStartValidationPipe from './pipes/deploy.start.pipe'
+import DeployCreateTeamAccessGuard from './guards/deploy.create.team-access.guard'
+import DeployTeamAccessGuard from './guards/deploy.team-access.guard'
+import DeployCopyValidationInterceptor from './interceptors/deploy.copy.interceptor'
+import DeployCreateValidationInterceptor from './interceptors/deploy.create.interceptor'
+import DeleteDeploymentValidationInterceptor from './interceptors/deploy.delete.interceptor'
+import DeployPatchValidationInterceptor from './interceptors/deploy.patch.interceptor'
+import DeployStartValidationInterceptor from './interceptors/deploy.start.interceptor'
+
+const DeploymentId = () => Param('deploymentId')
+const InstanceId = () => Param('instanceId')
 
-@UseGuards(JwtAuthGuard)
-@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor)
-@UseFilters(HttpExceptionFilter)
-@Controller('deploy')
+const ROUTE_DEPLOYMENTS = 'deployments'
+const ROUTE_DEPLOYMENT_ID = ':deploymentId'
+const ROUTE_INSTANCES = 'instances'
+const ROUTE_INSTANCE_ID = ':instanceId'
+@Controller(ROUTE_DEPLOYMENTS)
+@ApiTags(ROUTE_DEPLOYMENTS)
+@UseGuards(JwtAuthGuard, DeployTeamAccessGuard)
+@UsePipes(
+  new ValidationPipe({
+    // TODO(@robot9706): Move to global pipes after removing gRPC
+    transform: true,
+  }),
+)
+@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor, CreatedWithLocationInterceptor)
 export default class DeployHttpController {
   constructor(private service: DeployService) {}
 
+  @Get()
+  @ApiOkResponse({
+    type: DeploymentDto,
+    isArray: true,
+  })
+  async getDeployments(@IdentityFromRequest() identity: Identity): Promise<DeploymentDto[]> {
+    return await this.service.getDeployments(identity)
+  }
+
+  @Get(ROUTE_DEPLOYMENT_ID)
+  @ApiOkResponse({
+    type: DeploymentDetailsDto,
+    isArray: true,
+  })
+  async getDeploymentDetails(@DeploymentId() deploymentId: string): Promise<DeploymentDetailsDto> {
+    return await this.service.getDeploymentDetails(deploymentId)
+  }
+
+  @Get(`${ROUTE_DEPLOYMENT_ID}/events`)
+  @ApiOkResponse({ type: DeploymentEventDto, isArray: true })
+  async getDeploymentEvents(@DeploymentId() deploymentId: string): Promise<DeploymentEventDto[]> {
+    return await this.service.getDeploymentEvents(deploymentId)
+  }
+
+  @Get(`${ROUTE_DEPLOYMENT_ID}/${ROUTE_INSTANCES}/${ROUTE_INSTANCE_ID}`)
+  @ApiOkResponse({ type: InstanceDto })
+  async getInstance(@InstanceId() instanceId: string): Promise<InstanceDto> {
+    return await this.service.getInstance(instanceId)
+  }
+
+  @Get(`${ROUTE_DEPLOYMENT_ID}/${ROUTE_INSTANCES}/${ROUTE_INSTANCE_ID}/secrets`)
```

Comment:
I love the way how we use constants, but right here when I check this controller I will never be able to figure out what path this is. Just a comment.